src/azstoragetorch/io.py

```diff
@@ -229,9 +284,74 @@ def _compute_new_position(self, offset: int, whence: int) -> int:
             return self._client.get_blob_size() + offset
         raise ValueError(f"Unsupported whence: {whence}")
 
+    def _flush(self) -> None:
+        self._check_for_stage_block_exceptions(wait=False)
+        self._flush_write_buffer()
+        self._check_for_stage_block_exceptions(wait=True)
+
+    def _flush_write_buffer(self) -> None:
+        if self._write_buffer:
+            futures = self._client.stage_blocks(memoryview(self._write_buffer))
+            self._all_stage_block_futures.extend(futures)
+            self._in_progress_stage_block_futures.extend(futures)
+            self._write_buffer = bytearray()
+
+    def _write(self, b: _SUPPORTED_WRITE_TYPES) -> int:
+        self._check_for_stage_block_exceptions(wait=False)
+        write_length = len(b)
+        self._write_buffer.extend(b)
+        if len(self._write_buffer) >= self._WRITE_BUFFER_SIZE:
+            self._flush_write_buffer()
+        self._position += write_length
+        return write_length
+
+    def _commit_blob(self) -> None:
+        self._flush()
+        block_ids = [f.result() for f in self._all_stage_block_futures]
+        self._raise_if_duplicate_block_ids(block_ids)
+        self._client.commit_block_list(block_ids)
+
+    def _raise_if_duplicate_block_ids(self, block_ids: str) -> None:
+        # An additional safety measure to ensure we never reuse block IDs within a BlobIO instance. This
+        # should not be an issue with UUID4 for block IDs, but that may not always be the case if
+        # block ID generation changes in the future.
+        if len(block_ids) != len(set(block_ids)):
+            raise RuntimeError(
```

Comment:
Do we want to raise `FatalBlobIOWriteError` instead?

Reply:
I was on the fence about this one when I chose `RuntimeError`. I could go either way especially since if a customer ever got this error, the best option (assuming there is not consistent collisions) is to just retry it. I went with `RuntimeError` to be more indicative that this was not expected and should never be hit (i.e. panic as this is definitely a bug). Thoughts? I'm good with changing it if you like that more.

Reply:
I didn't have a strong opinion so this is fine.